packages/cosmic-swingset/scripts/clean-core-eval.js

```diff
@@ -61,11 +61,15 @@ export const main = async (argv, { readFile, stdout }) => {
 
 if (isEntrypoint(import.meta.url)) {
   /* global process */
-  main([...process.argv], {
-    readFile: (await import('fs/promises')).readFile,
-    stdout: process.stdout,
-  }).catch(err => {
-    process.exitCode = 1;
-    console.error(err);
-  });
+  import('fs/promises')
+    .then(fsp =>
+      main([...process.argv], {
+        readFile: fsp.readFile,
+        stdout: process.stdout,
+      }),
+    )
+    .catch(err => {
+      process.exitCode = 1;
+      console.error(err);
+    });
```

Comment:
```suggestion
  E.when(
    import('fs/promises'),
    fsp =>
      main([...process.argv], {
        readFile: fsp.readFile,
        stdout: process.stdout,
      }),
    ),
    err => {
      process.exitCode = 1;
      console.error(err);
    });
```
Btw, I do realize that I'm making these suggestions on where you've already improved the code to use `.then` or `.catch` rather than `await`. And that an `await` itself also fails to leave behind any diagnostic info (track-turns, deep stacks). At some point we should explore whether there are any debugging / tracing hooks we should ask machines (e.g. XS) to provide. Likely, whatever those are, it would also overcome this problem with `.then`/`.catch`.